source/_integrations/pid_controller.markdown

```diff
@@ -0,0 +1,152 @@
+---
+title: PID Controller
+description: Creates a PID controller to regulate a sensor value with a number-entity output.
+ha_category:
+  - Number
+ha_release: '2023.05'
+ha_iot_class: Local Polling
+ha_domain: pid_controller
+ha_platforms:
+  - number
+ha_integration_type: integration
+---
+
+The `pid_controller` platform is a `number`. It can be used to control a number entity output to regulate a sensor value to a certain setpoint. The value of the number entity is the setpoint. As a sensor, any numerical senosor entity can be used. If two sensors are configured, the PID controller will act as a differential controller, using the difference between the two sensor values as input signal.
+The value for the output number entity will be calculated using the Proportional–Integral–Derivative algorithm (PID, See [https://en.wikipedia.org/wiki/PID_controller]). The implementation of the PID controller contains bumpless operation, and is prevented against integral windup by clipping of the output value to the minimum and maximum of the corresponding output number entity. 
+This controller is typically usefull in regulated systems, for example regulate the speed of a water pump in a heat collector to keep the temperature difference between the outgoing and incomming waterstream on a certain level, so that the heat collector will perform optimal.
+
+## Configuration
+{% include integrations/config_flow.md %}
+{% configuration_basic %}
+Name:
+  description: The name that the PID controller should have. You can change it again later.
+Output number entity:
+  description: Output number entity. The output of the regulator will be clipped to the min- and max value of this output number.
+Input sensor entity:
+  description: Numerical sensor entity, used for input signal.
+Optional secondary input sensor entity:
+  description: Secondary input sensor. If selected, the regulator will work in differential mode.
+Proportional gain factor (Kp):
+  description: Proportional gain factor, directly gaining the error to compensate the fault (Kp).
+Integration factor (Ki):
+  description: Integration factor, reducing offset fault over time (Ki).
+Differential factor (Kd):
+  description: Differential factor, damping the overshoot (Kd).
+Controller direction: 
+  description: When direct, output will increase to decrease fault, when reverse output will decrease to decrease fault.
```

Comment:
```suggestion
  description: When 'direct', the output will increase to decrease fault. When 'reverse', the output will decrease to decrease fault.
```

Reply:
I'm probably not the target audience of this text and maybe this is entirely clear to people who know their way around PID and PWM. How do I know what values to pick for kp, ki, and kd?

Reply:
Hm, value highly depends on the application of the regulator. I think those looking for a PID regulator know what they set up, and also they have an idea what they want to put there. But mayb I can think out some suggestions:
- For kp, start with a small number (1) and gradually make it bigger if you see that the direct reaction of the controller is too low.
- For ki, start with a very small number (0.01). If you see that the reaction  over time is only slowly rising, than increase it.
- For kd, keep this number to 0 until kp and ki are set. Now you can use the kd to prevent the regulator from overshoot. Only increase in small steps.